docs/source/query-syntax.rst

```diff
@@ -0,0 +1,133 @@
+.. _query-syntax:
+
+Search
+======
+
+The MLflow UI and API support searching runs within a single experiment or a group of experiments
+using a search filter API. This API is a simplified version of the SQL ``WHERE`` clause.
+
+.. contents:: Table of Contents
+  :local:
+  :depth: 3
+
+Syntax
+------------------
+
+A search filter can be one or more expressions joined by the ``AND`` keyword.
+The syntax does not support ``OR``. Each expression has three parts: an identifier on
+the left-hand side (LHS), a comparator, and constant on the right-hand side (RHS).
+
+Identifier
+^^^^^^^^^^
+
+Required in the LHS of a search expression. Signifies an entity to compare against. An identifier has two
+parts separated by a period: the type of the entity and the name of the entity. 
+The type of the entity is ``metrics``, ``params``, or ``tags``. For example: ``metrics.accuracy``.
+
+Entity Name Contains Special Characters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When a metric, parameter, or tag name contains special characters like hyphen, space, period, and so on,
+enclose the entity name in double quotes.
+
+.. rubric:: Examples
+
+.. code-block:: sql
+
+  params."model-type" = "LogisticRegression"
+
+.. code-block:: sql
+
+  metrics."error rate" <= 0.01
+
+
+Entity Name Starts with a Number
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Unlike SQL syntax for column names, MLflow allows logging metrics, parameters, and tags with names
+that have a leading number. If an entity name contains a leading number, enclose the entity name in quotes. 
+For example:
+
+.. code-block:: sql
+
+  metrics."2019-04-02 error rate" <= 0.35
+
+
+Comparator
+^^^^^^^^^^
+
+There are two classes of comparators: numeric and string.
+
+- Numeric comparators (``metrics``): ``=``, ``!=``, ``>``, ``>=``, ``<``, and ``<=``.
+- String comparators (``params`` and ``tags``): ``=`` and ``!=``.
+
+Constant
+^^^^^^^^
+
+The search syntax requires the RHS of the expression to be a constant. The type of the constant
+depends on LHS.
+
+- If LHS is a metric, the RHS must be an integer or float number.
+- If LHS is a parameter or tag, the RHS must be a string constant enclosed in single or double quotes.
+
+
+Example Expressions
```

Comment:
Does it make sense to pull these examples up top before introducing the syntax? With those special cases, some examples are already introduced.

Reply:
Hmm. I see your point. Would you be OK with only leaving the identifier in each example?